benchmark/displace.py

```diff
@@ -0,0 +1,69 @@
+"""
+Eager implementation of displacement generation benchmark
+
+Created on Wed Jun 10 12:44:33 2020
+
+@author: Henry Liu
+"""
+import numpy as np
+import scipy as sc
+import tensorflow as tf
+
+import utils
+
+
+def gen_displace_scipy(N):
+    a = np.diag(np.sqrt(np.arange(1, N, dtype=np.complex64)), k=1)
+    a_dag = np.conjugate(a.transpose())
+    vexpm = np.vectorize(sc.linalg.expm, signature="(n,n)->(n,n)")
+
+    def displace(alphas):
+        alphas = np.expand_dims(np.expand_dims(alphas, -1), -1)
+        return vexpm(alphas * a_dag - np.conj(alphas) * a)
+
+    return displace
+
+
+def gen_displace(N):
+    a = utils.creation(N)
+    a_dag = tf.linalg.adjoint(a)
+
+    def displace(alphas):
+        # wrap for broadcasting [num, 1, 1]
+        alphas = tf.expand_dims(tf.expand_dims(alphas, -1), -1)
```

Comment:
I will add some docs - this is the same as reshape [num] -> [num, 1, 1] so that it can be broadcast with [num, N, N]. I thought this is slightly cleaner since num doesn't have to be specified explicitly:
```
tf.reshape(alphas, [alphas.shape[0], 1, 1)
```

Reply:
I thought that reshape would be better because then it doesn't care if the input is of shape [num] or [num,1] or [num,1,1], while expand dims relies on shape [num]. So I guess we can keep as it is, and just add a doc as you suggested? Depends how we will use it in the future. I had a similar function before and found that I want to be able to call it with different input shapes.

Reply:
Got it, sorry I misunderstood! I agree, that's more flexible, I'll switch to that